modules/ROOT/pages/dataweave-cookbook-csv-lookup.adoc

```diff
@@ -0,0 +1,78 @@
+= Look Up Data in a CSV File
+ifndef::env-site,env-github[]
+include::_attributes.adoc[]
+endif::[]
+
+This DataWeave example uses the `filter` function to return the country code stored in a CSV file to the specified input phone numbers.
+//LINK TO DW 1.0 EXAMPLES:
+include::partial$dataweave1-links.adoc[tag=dataweave1Examples]
+
+Given the following JSON input phone numbers as the payload:
+
+[source,JSON,linenums]
+----
+[
+   "54-112724555",
+   "1-6298765432"
+]
+----
+
+And the CSV file in a var called `country_codes` with the following values:
+
+[source,CSV,linenums]
+----
+CALLING_CODE,COUNTRY_CODE
+1,US
+7,RU
+54,AR
+20,EG
+32,BE
+33,FR
+505,NI
+506,CR
+1876,JM
+1905,CA
+1939,PR
+262262,RE
+262269,YT
+.,.
+.,.
+----
+
+The following DataWeave script reads the input payload phone numbers, and uses the `filter` function to search the country code in the variable `country_code` that match with the calling code. Subsequently, uses the `map` function to get the country code, and returns the results in an array of JSON objects:
```

Comment:
```suggestion
The following DataWeave script reads each phone number in the payload and uses the `filter` function to search for the calling code within the Mule variable `country_code` that matches the calling code in the phone number. Then the `map` function gets the country code that corresponds to the calling code and returns the results to an array of JSON objects.
```

Reply:
Note that it formerly said that `filter` matches the country code to the calling code, but if you look at `(vars.country_code filter ((item, index) -> item.CALLING_CODE == callingCode)`, you can see that it's filtering based on matching the calling code (not country code) in the CSV to the calling code in the phone number. The `map` function is actually mapping the calling code to the country code after the filter function finds the matching calling code.